R/expect_identical_linter.R

```diff
@@ -0,0 +1,74 @@
+#' Require usage of expect_identical(x, y) where appropriate
+#'
+#' At Google, [testthat::expect_identical()] should be the default/go-to function for
+#'   comparing an output to an expected value. `expect_true(identical(x, y))`
+#'   is an equivalent but unadvised method of the same test. Further,
+#'   [testthat::expect_equal()] should only be used when `expect_identical()`
+#'   is inappropriate, i.e., when `x` and `y` need only be *numerically
+#'   equivalent* instead of fully identical (in which case, provide the
+#'   `tolerance=` argument to `expect_equal()` explicitly). This also applies
+#'   when it's inconvenient to check full equality (e.g., names can be ignored,
+#'   in which case `ignore_attr = "names"` should be supplied to
+#'   `expect_equal()` (or, for 2nd edition, `check.attributes = FALSE`).
+#'
+#' NB: The linter allows `expect_equal()` in three circumstances:
```

Comment:
Not a fan of NB in documentation.
I'd a) drop the NB and leave the rest as-is or b) make an `@section Exceptions` to highlight that there are exceptions to the general rule stated above.